File: src/domain/bundledFonts.ts
// The same licensed TrueType faces are embedded in the Rust exporter.
export const BUNDLED_FONT_FACES = [
  {
    "family": "Inter",
    "weight": 400,
    "style": "italic"
  },
  {
    "family": "Inter",
    "weight": 700,
    "style": "italic"
  },
  {
    "family": "Inter",
    "weight": 400,
    "style": "normal"
  },
  {
    "family": "Inter",
    "weight": 600,
    "style": "normal"
  },
  {
    "family": "Inter",
    "weight": 700,
    "style": "normal"
  },
  {
    "family": "Playfair Display",
    "weight": 400,
    "style": "italic"
  },
  {
    "family": "Playfair Display",
    "weight": 700,
    "style": "italic"
  },
  {
    "family": "Playfair Display",
    "weight": 400,
    "style": "normal"
  },
  {
    "family": "Playfair Display",
    "weight": 600,
    "style": "normal"
  },
  {
    "family": "Playfair Display",
    "weight": 700,
    "style": "normal"
  },
  {
    "family": "Montserrat",
    "weight": 400,
    "style": "italic"
  },
  {
    "family": "Montserrat",
    "weight": 700,
    "style": "italic"
  },
  {
    "family": "Montserrat",
    "weight": 400,
    "style": "normal"
  },
  {
    "family": "Montserrat",
    "weight": 600,
    "style": "normal"
  },
  {
    "family": "Montserrat",
    "weight": 700,
    "style": "normal"
  },
  {
    "family": "Cormorant Garamond",
    "weight": 400,
    "style": "italic"
  },
  {
    "family": "Cormorant Garamond",
    "weight": 700,
    "style": "italic"
  },
  {
    "family": "Cormorant Garamond",
    "weight": 400,
    "style": "normal"
  },
  {
    "family": "Cormorant Garamond",
    "weight": 600,
    "style": "normal"
  },
  {
    "family": "Cormorant Garamond",
    "weight": 700,
    "style": "normal"
  },
  {
    "family": "Cinzel",
    "weight": 400,
    "style": "normal"
  },
  {
    "family": "Cinzel",
    "weight": 600,
    "style": "normal"
  },
  {
    "family": "Cinzel",
    "weight": 700,
    "style": "normal"
  },
  {
    "family": "Great Vibes",
    "weight": 400,
    "style": "normal"
  }
] as const;

export async function loadAlbumFonts(): Promise<void> {
  if (typeof document === 'undefined' || !document.fonts) return;
  await Promise.all(BUNDLED_FONT_FACES.map((face) => document.fonts.load(`${face.style} ${face.weight} 24px "${face.family}"`)));
}

File: src-tauri/src/export_engine/bundled_fonts.rs
// Licensed font faces and attribution: public/fonts/*-OFL.txt.
pub fn bundled_font(family: &str, weight: u16, italic: bool) -> Option<&'static [u8]> {
    let faces: &[(u16, bool, &[u8])] = match family.to_lowercase().as_str() {
        "inter" => &[
            (400, true, include_bytes!("../../../public/fonts/inter-400-italic.ttf")),
            (700, true, include_bytes!("../../../public/fonts/inter-700-italic.ttf")),
            (400, false, include_bytes!("../../../public/fonts/inter-400-normal.ttf")),
            (600, false, include_bytes!("../../../public/fonts/inter-600-normal.ttf")),
            (700, false, include_bytes!("../../../public/fonts/inter-700-normal.ttf")),
        ],
        "playfair display" => &[
            (400, true, include_bytes!("../../../public/fonts/playfairdisplay-400-italic.ttf")),
            (700, true, include_bytes!("../../../public/fonts/playfairdisplay-700-italic.ttf")),
            (400, false, include_bytes!("../../../public/fonts/playfairdisplay-400-normal.ttf")),
            (600, false, include_bytes!("../../../public/fonts/playfairdisplay-600-normal.ttf")),
            (700, false, include_bytes!("../../../public/fonts/playfairdisplay-700-normal.ttf")),
        ],
        "montserrat" => &[
            (400, true, include_bytes!("../../../public/fonts/montserrat-400-italic.ttf")),
            (700, true, include_bytes!("../../../public/fonts/montserrat-700-italic.ttf")),
            (400, false, include_bytes!("../../../public/fonts/montserrat-400-normal.ttf")),
            (600, false, include_bytes!("../../../public/fonts/montserrat-600-normal.ttf")),
            (700, false, include_bytes!("../../../public/fonts/montserrat-700-normal.ttf")),
        ],
        "cormorant garamond" => &[
            (400, true, include_bytes!("../../../public/fonts/cormorantgaramond-400-italic.ttf")),
            (700, true, include_bytes!("../../../public/fonts/cormorantgaramond-700-italic.ttf")),
            (400, false, include_bytes!("../../../public/fonts/cormorantgaramond-400-normal.ttf")),
            (600, false, include_bytes!("../../../public/fonts/cormorantgaramond-600-normal.ttf")),
            (700, false, include_bytes!("../../../public/fonts/cormorantgaramond-700-normal.ttf")),
        ],
        "cinzel" => &[
            (400, false, include_bytes!("../../../public/fonts/cinzel-400-normal.ttf")),
            (600, false, include_bytes!("../../../public/fonts/cinzel-600-normal.ttf")),
            (700, false, include_bytes!("../../../public/fonts/cinzel-700-normal.ttf")),
        ],
        "great vibes" => &[
            (400, false, include_bytes!("../../../public/fonts/greatvibes-400-normal.ttf")),
        ],
        _ => return None,
    };
    let use_italic = italic && faces.iter().any(|(_, it, _)| *it);
    let target = if weight <= 500 { 400 } else { weight };
    faces.iter().filter(|(_, it, _)| *it == use_italic)
        .min_by_key(|(w, _, _)| w.abs_diff(target)).map(|(_, _, bytes)| *bytes)
}
